influxdb3_id/src/lib.rs

```diff
@@ -1,10 +1,14 @@
 use serde::Deserialize;
 use serde::Serialize;
+use serde_with::serde_as;
+use serde_with::DisplayFromStr;
+use std::fmt::Display;
 use std::sync::atomic::AtomicU32;
 use std::sync::atomic::Ordering;
 
-#[derive(Debug, Copy, Clone, Eq, PartialOrd, Ord, PartialEq, Serialize, Deserialize)]
-pub struct DbId(u32);
+#[serde_as]
+#[derive(Debug, Copy, Clone, Eq, PartialOrd, Ord, PartialEq, Serialize, Deserialize, Hash)]
+pub struct DbId(#[serde_as(as = "DisplayFromStr")] u32);
```

Comment:
That's correct. If we turn it into a vec though then we lose on the O(1) lookup for quite a few things as the id will not correspond to a position in a vec. It might make sense to have it serialize as a number if it's the value for a field and a string for the map key

Reply:
Yeah, so I think we should have one data structure for serialization, which is a vec of the objects, rather than a map. That way we can have all the IDs be ints. Then when you read the serialized data into an in-memory structure for fast lookups, you put it into a map, which should be by int. We should never being lookups against the serialized structure, I think